### PR TITLE
For #11940: Disable auto report on the AndroidManifest

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt [hunk 1]
```diff
@@ -120,7 +120,13 @@ class SentryService(
     @VisibleForTesting
     internal fun initSentry() {
         SentryAndroid.init(applicationContext) { options ->
+            // Disable uncaught no native exceptions from being reported.
             options.enableUncaughtExceptionHandler = false
+            // Disable uncaught native exceptions from being reported.
+            // Sentry don't have a way to disable uncaught native exceptions from being reported.
+            // As a fallback we had to disabled all native integrations.
```
Comment (csadilek) [inline components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt:127]: nit: had to disable.

File: components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt [hunk 2]
```diff
@@ -120,7 +120,13 @@ class SentryService(
     @VisibleForTesting
     internal fun initSentry() {
         SentryAndroid.init(applicationContext) { options ->
+            // Disable uncaught no native exceptions from being reported.
```
Comment (csadilek) [inline components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt:123]: nit: non-native maybe?

Comment (csadilek) [inline components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt:123]: I would point out here in the docs that we have our own uncaught exception handler so we don't need Sentry's default one.

**Review by csadilek — APPROVED** [review]
👍 🤞 

Some comment suggestions below.

File: components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt (hunk 1 above)
Comment (Amejia481) [reply components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt:127]: Thanks!!

File: components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt (hunk 2 above)
Comment (Amejia481) [reply components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt:123]: 👍🏽

Comment (Amejia481) [reply components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt:123]: 👍🏽

**Pull request merged**